src/execution_plan/ops/op_apply.c

```diff
@@ -1,97 +1,102 @@
 /*
- * Copyright Redis Ltd. 2018 - present
- * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
- * the Server Side Public License v1 (SSPLv1).
+ * Copyright FalkorDB Ltd. 2023 - present
+ * Licensed under the Server Side Public License v1 (SSPLv1).
  */
 
 #include "op_apply.h"
 #include "../execution_plan_build/execution_plan_util.h"
 
-/* Forward declarations. */
+// forward declarations
 static OpResult ApplyInit(OpBase *opBase);
 static Record ApplyConsume(OpBase *opBase);
 static OpResult ApplyReset(OpBase *opBase);
 static OpBase *ApplyClone(const ExecutionPlan *plan, const OpBase *opBase);
 static void ApplyFree(OpBase *opBase);
 
-OpBase *NewApplyOp(const ExecutionPlan *plan) {
-	Apply *op = rm_malloc(sizeof(Apply));
+OpBase *NewApplyOp
+(
+	const ExecutionPlan *plan
+) {
+	Apply *op = rm_calloc(1, sizeof(Apply));
 
-	op->r            = NULL;
-	op->op_arg       = NULL;
-	op->records      = NULL;
-	op->rhs_branch   = NULL;
-	op->bound_branch = NULL;
-
-	// Set our Op operations
+	// set our Op operations
 	OpBase_Init((OpBase *)op, OPType_APPLY, "Apply", ApplyInit, ApplyConsume,
 			ApplyReset, NULL, ApplyClone, ApplyFree, false, plan);
 
 	return (OpBase *)op;
 }
 
-static OpResult ApplyInit(OpBase *opBase) {
+static OpResult ApplyInit
+(
+	OpBase *opBase
+) {
 	ASSERT(opBase->childCount == 2);
 
 	Apply *op = (Apply *)opBase;
-	// the op's bound branch and optional match branch have already been
-	// built as the Apply op's first and second child ops, respectively
 	op->records      = array_new(Record, 1);
 	op->rhs_branch   = opBase->children[1];
 	op->bound_branch = opBase->children[0];
 
 	// locate branch's Argument op tap
-	op->op_arg = (Argument *)ExecutionPlan_LocateOp(op->rhs_branch,
+	op->args = array_new(Argument*, 1);
```

Comment:
get the op argements from the constructor
and add test for nested apply

Reply:
It would be wise to make this same change to all of our `Apply` variations e.g. `OR/AND Apply Multiplexer`, `Semi Apply` and `Anti Semi Apply`.

The only possible issue I see with this request is identifying the taps at compile time, there might be run-time changes which modify the `Argument` taps (still needs to be verified).